pl_engine/pl_server/src/main/java/com/cubrid/plcsql/compiler/ast/BuiltinFuncCall.java

```diff
@@ -30,18 +30,44 @@
 
 package com.cubrid.plcsql.compiler.ast;
 
+import com.cubrid.plcsql.compiler.ast.loopOpt.SqlUse;
 import com.cubrid.plcsql.compiler.type.Type;
 import org.antlr.v4.runtime.ParserRuleContext;
 
-public abstract class BuiltinFuncCall extends Expr {
+public abstract class BuiltinFuncCall extends Expr implements SqlUse {
+
+    public StmtLoop containerLoop;
+    public int sqlSerialNo;
```

Comment:
```suggestion
    private StmtLoop containerLoop;
    private int sqlSerialNo;
```

setter가 존재하므로 getter도 추가하여, public 보다는 private으로 선언하여 관리를 용이하게 하는 것이 좋을 것 같습니다.

Reply:
BuiltinFuncCall 타입 변수에 containerLoop 을 set 하는 코드가 있어서 setContainerLoop 는 있으나, 읽을 때는 구체적인 하위 클래스 변수에서만 읽어서 getter 가 없습니다. sqlSerialNo는 비슷한 이유로 getter는 있으나 setter는 없습니다.

Reply:
public으로 선언되어 있기 때문에, 외부에서 setter를 사용하지 않더라도 해당 멤버 변수에 접근할 수 있지 않나요?

Reply:
BuiltinFuncCall 이 아니라 구체적인 클래스를 통해서 접근할 때 method 호출보다는 field 접근이 약간이라도 성능면에서 좋을 것 같아서 그냥 두는 것이 좋을 것 같습니다. 
단, sqlSerialNo 는 생성자에서 정해지므로 final 로 해 둘 수 있네요.

Reply:
성능 때문이시라면... [JVM의 JIT(Just In Time) 컴파일러 덕분에 생성되는 머신 코드(바이트코드 x)는 똑같아서 상관없을 걸로 판단됩니다.](https://stackoverflow.com/questions/23931546/java-getter-and-setter-faster-than-direct-access)
추후에 리팩토링을 계획할 계획이 있으시다면, 외부에서 접근하지 못하도록 접근지정자를 수정하는 것이 좋을 것 같습니다.

Reply:
좋은 자료 감사합니다. getter 에 대해서 최종적으로 생성되는 machine code 에는 차이가 없다는 내용이네요.

그런데, 성능에는 차이가 없이 동일하다면, 
객체 외부에서 읽고 쓸 필요가 있는 클래스 필드 f 에 대하여
```
public T f;
```
와 같이 쓰는 것과 
```
private T f;
public T getF() {
    return f;
}
public void setF(T f) {
    this.f = f;
}
```
를 비교해 봤을 때 전자가 더 짧고 간결해서 좋아보입니다. 
후자가 관리가 용이하다는 것이 어떤 이유에서 일까요.

Reply:
> public으로 선언되어 있기 때문에, 외부에서 setter를 사용하지 않더라도 해당 멤버 변수에 접근할 수 있지 않나요?
com.cubrid.plcsql.compiler.ast.StmtLoop 를 보시면, SqlUse (LoopOptimizable) 변수에 대해 container loop 를 set 하고 있습니다. SqlUse 나 LoopOptimizable 이 필드를 선언할 수 없는 interface 이다보니 setter 를 통해 containerLoop 필드를 set 하도록 하고 있습니다.

Reply:
관리가 용이하다는 것은 추후 변경이 필요할 때, 수정이 편리하다는 것을 의미합니다.
하지만 이야기를 나눈 결과, 추후 변경될 가능성이 낮기 때문에 현 상태를 유지하는 것에 대해서 이해했습니다.